requirements.txt

```diff
@@ -3,17 +3,15 @@ setuptools>34.3.0
 wheel
 twine
 flake8
-nose
 nose-progressive
-coverage
-mock
-pytz
-mypy==0.770
-lxml # needed by mypy HTML coverage reporting
-pylint
-# sqlalchemy-vertica-python==0.4.5
-sqlalchemy-stubs>=0.3
+
```

Comment:
I needed to pull nose out to an extra so we could include it during integration testing without including everything here.  At that point, it also made sense to pull out coverage and mock and mypy to match the pattern.  pylint and pytz appear to be old cruft that's no longer required.

The rest of the stuff here is:

1. things we need on the command-line but never in CI (flake8, nose-progressive)
2. stuff we need at package publishing time (wheel, twine)
3. numpy-stubs (see note below)

Number one and two could be dealt with via an extra - assuming we haven't taken extras too far already, maybe that's the way to go?